tests/bits/step-51p.cc

```diff
@@ -343,10 +343,10 @@ namespace Step51
     typename FunctionMap<dim>::type boundary_functions;
     Solution<dim> solution_function;
     boundary_functions[0] = &solution_function;
-    VectorTools::project_boundary_values (dof_handler,
-                                          boundary_functions,
-                                          QGauss<dim-1>(fe.degree+1),
-                                          constraints);
+    VectorTools::project_boundary_values<dim,double> (dof_handler,
+                                                      boundary_functions,
```

Comment:
the reason why you need to specify the `number` type here and elsewhere is because of the use of `FunctionMap::type` which makes `number` non-deducible for a compiler. You can avoid all of these changes to the testsuite if you make `number` deducible to the compiler.